paperqa/agents/models.py

```diff
@@ -0,0 +1,520 @@
+        model = OpenAILLMModel(config={"model": llm_model, "temperature": 0.1})
+        chain = model.make_chain(
+            AsyncOpenAI(), prompt="{question}\n\n{answer}", system_prompt=sys_prompt
+        )
+        result = await chain({"question": self.answer.question, "answer": self.answer.answer})  # type: ignore[call-arg]
+        return result.text.strip()
+
+
+class TimerData(BaseModel):
+    # NOTE: use lambda because asyncio.get_running_loop can throw: RuntimeError: no running event loop
+    start_time: float = Field(
+        default_factory=lambda: asyncio.get_running_loop().time()  # noqa: FURB111
+    )
+    durations: list[float] = Field(default_factory=list)
+
+
+class SimpleProfiler(BaseModel):
+    """Basic profiler with start/stop and named timers.
+
+    The format for this logger needs to be strictly followed, as downstream google
+    cloud monitoring is based on the following
+    # [Profiling] {**name** of timer} | {**elapsed** time of function} | {**__version__** of PaperQA}
+    """
+
+    timers: dict[str, list[float]] = {}
+    running_timers: dict[str, TimerData] = {}
+    uid: UUID = Field(default_factory=uuid4)
+
+    @asynccontextmanager
+    async def timer(self, name: str):
+        start_time = asyncio.get_running_loop().time()
+        try:
+            yield
+        finally:
+            end_time = asyncio.get_running_loop().time()
+            elapsed = end_time - start_time
+            self.timers.setdefault(name, []).append(elapsed)
+            logger.info(
+                f"[Profiling] | UUID: {self.uid} | NAME: {name} | TIME: {elapsed:.3f}s | VERSION: {__version__}"
+            )
+
+    def start(self, name: str) -> None:
+        try:
+            self.running_timers[name] = TimerData()
+        except RuntimeError:  # No running event loop (not in async)
+            self.running_timers[name] = TimerData(start_time=time.time())
+
+    def stop(self, name: str):
+        timer_data = self.running_timers.pop(name, None)
+        if timer_data:
+            try:
+                t_stop: float = asyncio.get_running_loop().time()
+            except RuntimeError:  # No running event loop (not in async)
+                t_stop = time.time()
+            elapsed = t_stop - timer_data.start_time
+            self.timers.setdefault(name, []).append(elapsed)
+            logger.info(
+                f"[Profiling] | UUID: {self.uid} | NAME: {name} | TIME: {elapsed:.3f}s | VERSION: {__version__}"
+            )
+        else:
+            logger.warning(f"Timer {name} not running")
+
+    def results(self) -> dict[str, dict[str, float]]:
+        result = {}
+        for name, durations in self.timers.items():
+            low = min(durations)
+            mean = sum(durations) / len(durations)
+            max_ = max(durations)
+            result[name] = {
+                "low": low,
+                "mean": mean,
+                "max": max_,
+                "total": sum(durations),
+            }
```

Comment:
```suggestion
            mean = sum(durations) / len(durations)
            result[name] = {
                "low": min(durations),
                "mean": mean,
                "max": max(durations),
                "total": sum(durations),
            }
```

Imo sometimes less locals is more readable